src/main/java/com/google/devtools/build/lib/query2/TransitionsOutputFormatterCallback.java

```diff
@@ -128,7 +132,9 @@ public void processOutput(Iterable<ConfiguredTarget> partialResult) throws Inter
                     hostConfiguration,
                     /*aspect=*/ null,
                     configConditions,
-                    /*toolchainLabels=*/ ImmutableSet.of(),
+                    toolchainContext == null
+                        ? ImmutableSet.of()
+                        : toolchainContext.resolvedToolchainLabels(),
```

Comment:
Isn't it enough to just provide the UnloadedToolchainContext here and avoid the extra dependency?

If we *know* the full ToolchainContext is already available so doesn't incur extra time or memory, I guess the difference is negligible. But I'm inclined not to scope this object more than is actually needed, perhaps with a comment about the ability to take it farther if ever necessary.

@juliexxia  - do you foresee deeper reporting in ToolchainOutputFormatterCallback that would in fact want to dig into methods available on ToolchainContext but not UnloadedToolchainContext? (i.e. properties of the toolchain *target* instead of just its label)?

Reply:
We don't actually have an UnloadedToolchainContext here: it was created in ConfiguredTargetFunction but doesn't escape.

This change is the same thing that ConfiguredTargetFunction is now doing to get an UnloadedToolchainContext, so at this point it will always be available in Skyframe.

Reply:
Unless you mean to just change the type to UnloadedToolchainContext, that would be fine, ToolchainContext is just an interface exposing parts of that class.

Reply:
Ah, that makes sense - thanks for the explanation and apologies for being confused about the exact sequence.

With that said yes, I see no reason to artificially restrict the scope.